projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMFloatCompareNode.java

```diff
@@ -54,42 +54,71 @@ private static boolean areOrdered(float... vals) {
     public abstract static class LLVMFloatOltNode extends LLVMFloatCompareNode {
         @Specialization
         public boolean executeI1(float val1, float val2) {
-            return areOrdered(val1, val2) && val1 < val2;
+            if (val1 < val2) {
+                assert areOrdered(val1, val2);
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
IMHO, the latter (branching control flow). Profiling the input condition of an `if` is a convenient way to do that but the profile still belongs to the `if` (and its node), not the condition.